src/Hazelcast.Net/Sql/HazelcastSqlException.cs

```diff
@@ -13,26 +13,109 @@
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        private HazelcastSqlException(string message, Exception innerException)
+            : base(message, innerException)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        private HazelcastSqlException(Guid clientId, Guid memberId, int errorCode, string message)
+            : base(message)
         {
             ClientId = clientId;
+            MemberId = memberId;
             ErrorCode = errorCode;
         }
 
-        internal HazelcastSqlException(Guid clientId, SqlErrorCode errorCode, string message) : this(clientId, (int)errorCode, message)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        internal HazelcastSqlException(Guid clientId, SqlErrorCode errorCode, string message)
+            : this(clientId, Guid.Empty, (int)errorCode, message)
         { }
 
-        internal HazelcastSqlException(SqlError error) : this(error.OriginatingMemberId, error.Code, error.Message)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        internal HazelcastSqlException(Guid clientId, SqlError error)
+            : this(clientId, error.OriginatingMemberId, error.Code, error.Message)
         { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class with serialized data.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="StreamingContext"/> that contains contextual information about the source or destination.</param>
+        protected HazelcastSqlException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+            ClientId = info.GetGuid(nameof(ClientId));
+            MemberId = info.GetGuid(nameof(MemberId));
+            ErrorCode = info.GetInt32(nameof(ErrorCode));
+        }
+
+        /// <inheritdoc />
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            if (info == null) throw new ArgumentNullException(nameof(info));
+
+            info.AddValue(nameof(ClientId), ClientId);
+            info.AddValue(nameof(MemberId), MemberId);
+            info.AddValue(nameof(ErrorCode), ErrorCode);
+            base.GetObjectData(info, context);
+        }
+
+        /// <summary>
+        /// Get the identifier of the <see cref="IHazelcastClient"/> that caused the exception.
```

Comment:
Are we doing this, if the error is not originated from the client, this is null, thing?

Reply:
Well, Actually, `Guid.Empty` which is 00000-00000-0000-000000 or something - probably worth documenting, then?

Reply:
Yeah, I guess it makes sense to document the memberId behavior